### PR TITLE
E2E - Sig-autoscaling: Refactor the Autoscaling utils

### LINKED ISSUE #121892: E2E - Sig-autoscaling: Refactor the Autoscaling utils to use `gomega.Consistenly`  according to e2e test framework guidance
### What happened?

Originally from https://github.com/kubernetes/kubernetes/pull/121861#discussion_r1392089576

There was a bug https://github.com/kubernetes/kubernetes/issues/121860 that caused test to fail, and `gomega.Consistenly` was  used to replace the code but requires more updates to the testing code to be used as intended.

>When using gomega.Consistently, you are making a statement about some observable fact. The callback should only deliver that fact, the statement then should be (no pun intended) in the Should. That way you get all the advantages listed in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md#polling-and-timeouts.

> You also need to check error handling. Does `rc.GetReplicas` fail on errors? It would be better to return an error and let `Consistently` and `HandleRetry` deal with it.

 It fails hard with in its helper function after making the API call. If I understand correctly that isn't following the pattern suggested in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md#polling-and-timeouts. either. 
 https://github.com/kubernetes/kubernetes/blob/1bb2773153d5d7395d445feb23171ad8b84afb13/test/e2e/framework/autoscaling/autoscaling_utils.go#L444-L452

### What did you expect to happen?

Tests to follow recommended guidance.

### How can we reproduce it (as minimally and precisely as possible)?

See  discussion in https://github.com/kubernetes/kubernetes/pull/121861#discussion_r1392089576

### Anything else we need to know?

/kind cleanup
/sig autoscaling
/sig testing

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>
na

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Tests to follow recommended guidance.

#### Which issue(s) this PR fixes:

Fixes #121892 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jsturtevant** [comment]
running it again to make sure changes are stable since it failed once

/test pull-kubernetes-e2e-autoscaling-hpa-cpu

File: test/e2e/framework/autoscaling/autoscaling_utils.go [hunk 1]
```diff
@@ -368,25 +364,20 @@ func (rc *ResourceConsumer) sendConsumeCPURequest(ctx context.Context, millicore
...
+			return err
 		}
-		return true, nil
-	})
-
-	// Test has already finished (ctx got canceled), so don't fail on err from PollUntilContextTimeout
-	// which is a side-effect to context cancelling from the cleanup task.
-	if ctx.Err() != nil {
-		return
-	}
```
[the remark this answers is not in the document]
Comment (carlory) [reply test/e2e/framework/autoscaling/autoscaling_utils.go:380]: I forgot it is running in a goroutine thread. we need ignore the ctx cancel case, and I revert it back now.

**Comment by carlory** [comment]
/cc @pohly

please review again. thank you.

**Review by pohly — APPROVED** [review]
/lgtm
/approve

Thanks, and sorry for the delay.

**Pull request merged**